src/pages/signin/SAMLSignInPage/index.native.js

```diff
@@ -19,39 +26,61 @@ const defaultProps = {
     credentials: {},
 };
 
-const renderLoading = () => <FullScreenLoadingIndicator />;
+// const renderLoading = () => <FullScreenLoadingIndicator />;
 
 function SAMLSignInPage({credentials}) {
     const webViewRef = useRef();
     const samlLoginURL = `${CONFIG.EXPENSIFY.SAML_URL}?email=${credentials.login}&referer=${CONFIG.EXPENSIFY.EXPENSIFY_CASH_REFERER}`;
-
     /**
-    * Handles in-app navigation once we get a response back from Expensify
-    *
-    * @param {String} params.type
-    * @param {String} params.url
-    */
+     * Handles in-app navigation once we get a response back from Expensify
+     *
+     * @param {String} params.type
+     * @param {String} params.url
+     */
     const handleNavigationStateChange = useCallback(
         ({type, url}) => {
             const searchParams = new URLSearchParams(new URL(url).search);
             if (searchParams.has('shortLivedAuthToken')) {
                 const shortLivedAuthToken = searchParams.get('shortLivedAuthToken');
                 Session.signInWithShortLivedAuthToken(credentials.login, shortLivedAuthToken);
-                return;
+                Navigation.navigate(ROUTES.HOME);
```

Comment:
I don't think we need the `return;` or `Navigation.navigate(ROUTES.HOME)` here actually. `Session.signInWithShortLivedAuthToken` should take care of any necessary redirects